msu/hooks/states/world_state.nut

```diff
@@ -331,3 +332,27 @@
 		return __original(_mouse);
 	}
 });
+
+::MSU.QueueBucket.VeryLate.push(function() {
+	::MSU.MH.hook("scripts/states/world_state", function(q) {
+		q.onBeforeSerialize = @(__original) function( _out )
+		{
+			::MSU.System.Serialization.SerializationMetaData = _out.getMetaData();
+			::MSU.System.Serialization.MidOnBeforeSerialize = true;
+			__original(_out);
+			::MSU.System.Serialization.MidOnBeforeSerialize = false;
+		}
+
+		q.onBeforeDeserialize = @(__original) function( _in )
+		{
+			::MSU.System.Serialization.DeserializationMetaData = _in.getMetaData();
+			__original(_in);
+		}
+
+		q.onDeserialize = @(__original) function( _out )
+		{
+			__original(_out);
+			::MSU.System.Serialization.DeserializationMetaData = null;
```

Comment:
I once again ask for sanity check if we have ensured that this `onDeserialize` is the last function to actually run during deserialization and, therefore, it is safe to set this back to null here?

Reply:
Players are serialized before because roster etc. is done from the exe before all this. But are we sure there is no BB class that is serialized after world_state?

Reply:
I am reasonably sure, worst case we get a bug and fix it.